Add remote path settings to RemoteStoreSettings class

ashking94 opened the pull request:
<!--  Thanks for sending a pull request, here are some tips:

1. If this is a fix for an undisclosed security vulnerability, please STOP. All security vulnerability reporting and fixes should be done as per our security policy https://github.com/opensearch-project/OpenSearch/security/policy
2. If this is your first time, please read our contributor guidelines: https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md and developer guide https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md
3. Ensure you have added or ran the appropriate tests for your PR: https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md
-->

### Description
This is a follow up PR from https://github.com/opensearch-project/OpenSearch/pull/13155#discussion_r1565979927. In this PR, we have moved 2 settings - `cluster.remote_store.index.path.type` and `cluster.remote_store.index.path.hash_algorithm` to the RemoteStoreSettings class. 

### Check List
- ~[ ] New functionality includes testing.~
  - [x] All tests pass
- ~[ ] New functionality has been documented.~
  - ~[ ] New functionality has javadoc added~
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- ~[ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- ~[ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

sachinpkale commented:
Changes look good. Branch has multiple conflicts with `main`, most probably due to recent revert of: https://github.com/opensearch-project/OpenSearch/pull/13155

ashking94 commented:
@sachinpkale adding the reverted changes back as part of https://github.com/opensearch-project/OpenSearch/pull/13251. Once that is merged back, we should be good with the conflict part.

ashking94 commented:
Detect breaking changes is failing due to refactoring of some methods that are present in only main branch as of now and yet to be released. 
```
***! MODIFIED CLASS: PUBLIC org.opensearch.indices.IndicesService  (not serializable)
	===  CLASS FILE FORMAT VERSION: 55.0 <- 55.0
	---! REMOVED FIELD: PUBLIC(-) STATIC(-) FINAL(-) org.opensearch.common.settings.Setting<org.opensearch.index.remote.RemoteStoreEnums$PathHashAlgorithm> CLUSTER_REMOTE_STORE_PATH_HASH_ALGORITHM_SETTING
	---! REMOVED FIELD: PUBLIC(-) STATIC(-) FINAL(-) org.opensearch.common.settings.Setting<org.opensearch.index.remote.RemoteStoreEnums$PathType> CLUSTER_REMOTE_STORE_PATH_TYPE_SETTING
***  MODIFIED CLASS: PUBLIC org.opensearch.indices.RemoteStoreSettings  (not serializable)
	===  CLASS FILE FORMAT VERSION: 55.0 <- 55.0
	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) org.opensearch.common.settings.Setting<org.opensearch.index.remote.RemoteStoreEnums$PathHashAlgorithm> CLUSTER_REMOTE_STORE_PATH_HASH_ALGORITHM_SETTING
	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) org.opensearch.common.settings.Setting<org.opensearch.index.remote.RemoteStoreEnums$PathType> CLUSTER_REMOTE_STORE_PATH_TYPE_SETTING
	+++  NEW METHOD: PUBLIC(+) org.opensearch.index.remote.RemoteStoreEnums$PathHashAlgorithm getPathHashAlgorithm()
	+++  NEW METHOD: PUBLIC(+) org.opensearch.index.remote.RemoteStoreEnums$PathType getPathType()
```
We should ignore the "Detect breaking failure" changes here. cc @sachinpkale

ashking94 commented:
> ❕ Gradle check result for [61a0532](https://github.com/opensearch-project/OpenSearch/commit/61a0532f155b803dcedc6cf22182f755eed79d55): [UNSTABLE](https://build.ci.opensearch.org/job/gradle-check/37197/)
> 
> * **TEST FAILURES:**
> 
> ```
>       1 org.opensearch.remotestore.RemoteStoreStatsIT.testNonZeroPrimaryStatsOnNewlyCreatedIndexWithZeroDocs
>       1 org.opensearch.remotestore.RemoteStoreStatsIT.testDownloadStatsCorrectnessSinglePrimaryMultipleReplicaShards
> ```
> 
> Please review all [flaky tests](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) that succeeded after retry and create an issue if one does not already exist to track the flaky failure.

Flaky tests - https://github.com/opensearch-project/OpenSearch/issues/10983 and https://github.com/opensearch-project/OpenSearch/issues/10152

sachinpkale commented:
> Detect breaking changes is failing due to refactoring of some methods that are present in only main branch as of now and yet to be released.
> 
> ```
> ***! MODIFIED CLASS: PUBLIC org.opensearch.indices.IndicesService  (not serializable)
> 	===  CLASS FILE FORMAT VERSION: 55.0 <- 55.0
> 	---! REMOVED FIELD: PUBLIC(-) STATIC(-) FINAL(-) org.opensearch.common.settings.Setting<org.opensearch.index.remote.RemoteStoreEnums$PathHashAlgorithm> CLUSTER_REMOTE_STORE_PATH_HASH_ALGORITHM_SETTING
> 	---! REMOVED FIELD: PUBLIC(-) STATIC(-) FINAL(-) org.opensearch.common.settings.Setting<org.opensearch.index.remote.RemoteStoreEnums$PathType> CLUSTER_REMOTE_STORE_PATH_TYPE_SETTING
> ***  MODIFIED CLASS: PUBLIC org.opensearch.indices.RemoteStoreSettings  (not serializable)
> 	===  CLASS FILE FORMAT VERSION: 55.0 <- 55.0
> 	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) org.opensearch.common.settings.Setting<org.opensearch.index.remote.RemoteStoreEnums$PathHashAlgorithm> CLUSTER_REMOTE_STORE_PATH_HASH_ALGORITHM_SETTING
> 	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) org.opensearch.common.settings.Setting<org.opensearch.index.remote.RemoteStoreEnums$PathType> CLUSTER_REMOTE_STORE_PATH_TYPE_SETTING
> 	+++  NEW METHOD: PUBLIC(+) org.opensearch.index.remote.RemoteStoreEnums$PathHashAlgorithm getPathHashAlgorithm()
> 	+++  NEW METHOD: PUBLIC(+) org.opensearch.index.remote.RemoteStoreEnums$PathType getPathType()
> ```
> 
> We should ignore the "Detect breaking failure" changes here. cc @sachinpkale

Thanks for adding these details.
